careers/devops.markdown

```diff
@@ -0,0 +1,57 @@
+
+The successful candidate will be in the middle of it all, collaborating with developers at the forefront of making Haskell a groundbreaking, influential, top-tier programming language. The individual will improve existing infrastructure, work on creating a shared [CI](https://gitlab.haskell.org/ghc/ghc/-/tree/master/.gitlab) and performance benchmarking system across the Haskell ecosystem, shortened CI turnaround times, and improved ability to debug failures.
+
+# Responsibilities
+
+* Maintain CI (continuous integration) infrastructure for GHC (the Glasgow Haskell Compiler, the leading compiler for the Haskell language), as hosted on GHC's GitLab instance.
+* Work with the GHC team to improve CI efficiency, reliability, and resource utilization.
+* Design and implement expanding the CI infrastructure to include key components of the Haskell ecosystem.
+* Support other Haskell libraries and tools, including HLS (Haskell Language Server).
+* Support OS packaging of GHC and tools.
+* Documentation for the CI infrastructure.
+
+## Likely future responsibilities
+
+* Work with the [Haskell Infrastructure Team](https://github.com/haskell-infra/haskell-admins/).
+* Improvements and expansion of the GHC performance dashboard.
+
+# Qualifications
+
+* Demonstrated ability to manage and simplify complexity
+* Collaborate with a variety of stakeholders
+* Ability to choose between multiple, conflicting priorities
+* Excellent English written communication skills
+
+# Current Technology Stack
+
+* Most GHC services are hosted by a single machine running NixOS. This includes our PostreSQL database, GitLab and Grafana instances, our Docker registry, as well as various project-specific services. All of these services are provisioned via their associated Nix modules.
+* Our CI infrastructure builds upon GitLab Pipelines, using Docker for build isolation and reproducibility on Linux. A qualified applicant should be comfortable with both technologies.
+* We maintain a fleet of heterogeneous CI runners, covering Linux (AArch64, x86-64), Windows (x86-64), and Darwin (AArch64).
+
+# Bonus Points
+
+* Some famliarity with Haskell packaging.
+* Familiarity with platform-dependent software packaging (e.g. notarisation and code signing on Darwin).
+* Some systems programming experience.
+
+# Misc.
+
+This is a 100% remote position, reporting to the Executive Director of the Haskell Foundation, but taking technical direction from the GHC team (primarily Ben Gamari). Primary interactions will include people in US and European time zones, but can include coordination with people across the world at times.
+
+Haskell knowledge in not required.
+
+Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.
```

Comment:
![Screenshot-2022-01-13-19:50:15](https://user-images.githubusercontent.com/20933385/149382703-120fff2f-226a-4e07-942a-b5e829a7b647.png)

This is kinda offensive.

Of course, as in Google Summer of Code put it - they give equal opportunity to people - the same students for the same work done in different countries would receive ~1K in Somalia, & 5K in Switzerland. Indeed that is equal encouragement & opportunity. Since smart people can't exist/survive in Somalia, since if they educated themselves smart enough to work in GHC - as my friends told - people with higher education, the kind of polite smart people - are considered by military regimes a too valuable commodity to give up during the regime change & overall get considered a hazard to local military rulers. So friends told that on return to homeland they would be killed since they received higher education, a father of one of them, electrician, manages to not get killed, because the local rulers need electricity.
I know that the argument of the multiplication of a person's life on "The Economic Power Index of the country" works great on paper. Indeed people who managed to keep high morals & stay good people in Somalia, do not need money or help or encouragement.

---

Depending on my GPS location - I get drastically different salaries & work offers (in fact living here I get no offers at all, only automatic replies). I would uncover a secret, in Ukraine median salary is ~4323.6$. It is a year. Because, depending on the motivation, I'm not fancy & I can't get work (except rare invitations to do cryptocurrencies, which I begin to consider), I can survive here on those 360$ a month & would be ready to go damping that low, because I have high morals & disregard money reasoning. In addition to what is requested in the requirements I would also have the knowledge of Haskell tooling, and happen to Haskell on some level, so I guess that can be used as a starting point & people while live in Ukraine probably deserve it. But wait, there are countries where is an even cheaper workforce, so even those 360$/m can be saved depending on the geographic locations. There is a lot of economies that are worse than in Ukraine, where people live even in more peculiar situations.

---

I guess these are true facts nevertheless. If work is marked fully remote, at least do not advertise to a person that how much a person would earn for the same fully remote work depends on GPS location, because you would divert good specialists in poor countries & encourage people in rich countries to apply because of money. That is an organization's internal considerations and level of experience & bargaining the salary exists for it. The normal salary bargaining happens in the normal conditions. It is Ok to ask for "raw expectations" during interviews & establish the salary. Also probably it is better to not tell the maximum sum - because ... there seem to be no good logical paths of good outcomes from that. If the person would cost all $124K - you probably know that person & if they are interested - they would probably 1. be noticed on application 2. and would ask "how much HF got" in the chatroom or in the interview, if they need that much money.

In any situation: "The position was evaluated $124k, but I was chosen as the best candidate & highered at 48k" - that fact stays in the back of the head of the person during work, demotivating a person or can start processes of internal reasoning, in a long term that can play a role between during hard moments of high complexity tasts, high demand in work & overtimes - the person being motivated or not, if the person would know that position was $124k, but they do work for X$ - it can demotivate the person.

I suggest only the gradual Newtonian method of ballparking around the normal distribution deviation. People in capitalism seem to forget how to bargain. I still remember why the tradition of bargain existed, in childhood, it told me that character & people communication & people values worth more than prices, which frequently have a high degree of arbitrary.

I love honesty & mainly prefer it. But I am not monadic to honesty. Sometimes it is easier to pretend you do not know, or ascent to the level of to not care about what other people do/how they live, than to live, explicitly see &, live through it & feel it every day while you can't change it really for you, only you for future generations or society as a whole can change.

So, as a poor person from a poor country - I would ask to remove the mentioning that salary depends on "geographical location". And the upper bound of salary always seemed illogical to me, if you want to select people - set the minimal salary - that would show to applicants how much serious person needs to be to the requirements, the people that want too much to their level - is in itself a redflag to the quality of the person. The former one (country) - both parties know & it is a Polichinelle secret, the second one (max boundary) is illogical to mention, people anyway should not apply for profit reasons into nonprofit organizations, I think Haskell Foundation can find a DevOps for a reasonable cost without mentioning the upper bound.

Reply:
Thanks @Anton-Latukha. There's a lot there, but perhaps we can extract out this last piece: does it make more sense to mention a lower bound than an upper bound? I wrote a bit about how I was worried about stating an upper bound, but I then deleted it, as I wasn't sure -- and I think it's helpful to applicants to give an idea of what we're thinking about. We can give that idea via a lower bound instead. I don't know what's best here -- this is far from my area of expertise and experience -- but I think it's an interesting idea to consider.

Reply:
I used lower bound as an example, I not imply as it should be used.

I think that overall simply the "market competitive" salary indication should be used. But because standard wordings get plundered - the new wording needs to be constructed to imply the same.

Like, "we would be able to provide a normal market price for according experience in work".

Reply:
During job interviews simply reinstating that the position funding abilities are as in normal DevOps market - already enough.

Almost all people who would apply & you would review would land into 3k-9k range anyway, in that condition, the precise price can be negotiated in the end. I think most people who recently had earned DevOps salaries can wait a week or two for HF to ask a bunch of people on salary & compare them.

Also, the ["best bride marriage problem solution"](https://en.wikipedia.org/wiki/Secretary_problem) in optimization theory is good to know (evaluate people for 1/e interval of hiring time range & then start searching for the next best one), also the real-life better than "best bride", the people actually can be called back in time.

Reply:
So, that solution needs "the time range to hire".

Dice analytics in the salary report, report 53 days to hire a DevOps
![Screenshot-2022-01-14-23:02:24](https://user-images.githubusercontent.com/20933385/149584815-988776e0-51ad-4b88-b6d8-c8b796eaad77.png)
(source: Dice 2021 Tech Salary Report)